### PR TITLE
added feature to also include array as filter in GET /:name/rows route

### LINKED ISSUE #131: Enhance API to Support Multiple IDs in a Single Filter Parameter
**Is your feature request related to a problem? Please describe.**
The current version of the API only supports filters with single values, as demonstrated by the following URL: `http://localhost:8000/api/tables/customers/rows?_filters=CustomerId:1,country:Germany&_limit=100&_ordering=-id&_page=1`. This limitation prevents the use of multiple IDs in a single filter parameter.

**Describe the solution you'd like**
I propose enhancing the API's filtering capability to allow the use of multiple IDs within a single filter parameter. This would enable users to query data using an array of attributes such as IDs, as demonstrated in the following URL format: ```http://localhost:8000/api/tables/customers/rows?_filters=CustomerId:[1,2,3,4,5],country:Germany&_limit=100&_ordering=-id&_page=1``` 
OR 
```http://localhost:8000/api/tables/customers/rows?_filters=CustomerId:1,2,3,4,5,country:Germany&_limit=100&_ordering=-id&_page=1``` (comma separated).

Or, multiple non-PK fields could be provided:
`http://localhost:8000/api/tables/customers/rows?_filters=country:Germany,France&_limit=100&_ordering=-id&_page=1`

**Describe alternatives you've considered**
As an alternative, users could currently achieve similar results by making multiple API calls with individual ID filter values. However, this approach may lead to increased API usage and potentially slower data retrieval times, particularly when dealing with a substantial number of IDs. The other way is users can make the call as: `http://localhost:8000/api/tables/customers/rows/1,2,3,4,5` but this is limited to only PKs and not any other attribute.

**Additional context**
In scenarios where users need to retrieve data related to multiple specific IDs, such as the example URL provided, the ability to pass an array of IDs within a single filter parameter can significantly enhance the efficiency and effectiveness of data retrieval. This feature would contribute to an improved user experience and provide users with a more powerful querying tool within the application's API.

### PR DESCRIPTION
**Pull request by TahaKhanAbdalli**
After this merge the filters in GET `/:name/rows` can also include array e.g. `_filters=id:[1,2,3],firstName:John,lastName:Doe`

This fix has been sponsored by @IanMayo

fixes #131

### CODE REVIEW

File: core/src/controllers/rows.js [hunk 1]
```diff
@@ -77,8 +77,9 @@ const listTableRows = async (req, res) => {
   // e.g. ?_filters=name:John,age:20
   // will filter by name like '%John%' and age like '%20%'
   let filters = [];
+  const re = /(\w+:?(\[.*?\]|\w+)?)/g;
```
Comment (IanMayo) [inline core/src/controllers/rows.js:80]: could you include a comment immediately before you declare `re` to explain that we need to split the filters by `,` except when in an array, please @TahaKhanAbdalli

**Review by IanMayo — CHANGES_REQUESTED** [review]
Hello @TahaKhanAbdalli - this PR is _slightly_ more difficult to review because your linter is configured to use `"` instead of `'`.
Would you mind changing the config to use `'` string marker, and do a fresh save on this file?
The changes should then be easier to review.

**Review by IanMayo — APPROVED** [review]

**Comment by IanMayo** [comment]
Hello @thevahidal , any chance of review/merge on this PR please?

It's the last feature required in order for `Soul` to be compliant as a `REST` backend for [`react-admin`](https://marmelab.com/react-admin/)

**Comment by thevahidal** [comment]
@TahaKhanAbdalli Thanks for the PR, I think it's better if use a syntax like what we have for greater than, less than and ..., in those case we use a `gt`, `lt` lookup appended to the end of the field name, e.g. `id_gt=...`, `id_lt=...`.
I think it's good pattern to also use `in` for multiple lookup values, e.g. `id__in=...`.
@IanMayo What's your opinion on this?

File: core/src/controllers/rows.test.js [hunk 2]
```diff
@@ -102,19 +102,40 @@ describe('Rows Endpoints', () => {
...
+    );
+    expect(res.status).toEqual(200);
+    expect(res.body).toHaveProperty('data');
+    expect(res.body.data).toEqual(expect.any(Array));
+    expect(res.body.data.length).toEqual(2);
+  });
+
+  it('GET /tables/:name/rows: should return values if the provided ID matches the user ID.', async () => {
+    const res = await requestWithSupertest.get(
+      '/api/tables/users/rows?_filters=id:2'
```
Comment (thevahidal) [inline core/src/controllers/rows.test.js:133]: It's nice to add a test case for multiple lookups

**Comment by thevahidal** [comment]
@TahaKhanAbdalli There's also some conflicts needed to be handled.
Would you please bump of the minor version of Soul Core to `v0.6.0`

**Comment by IanMayo** [comment]
@thevahidal - sure, that strategy separates how API supports single vs multiple key values, which keeps things simple.

**Comment by IanMayo** [comment]
Hello @TahaKhanAbdalli - I support @thevahidal [suggestion](https://github.com/thevahidal/soul/pull/132#issuecomment-1681040130) to support multiple matching keys using the `__in` operator.

**Comment by thevahidal** [comment]
@IanMayo, the `__in=` is more of a nice-to-have feature and if `react-admin` is already using it the way @TahaKhanAbdalli implemented it, I'm ok to move on with this.

**Review by thevahidal — APPROVED** [review]
As we discuss it with @IanMayo, moving on with this PR is okay. Thanks, @TahaKhanAbdalli!

**Comment by thevahidal** [comment]
@all-contributors please add @thevahidal for review.

**Pull request merged**